### PR TITLE
[#1387] In-app Send feedback / contact support form

### LINKED ISSUE #1387: Send feedback / contact support form
## Motivation

Users have no in-app way to send feedback or report something that's broken. Right now the only channel is "find the GitHub repo and open an issue" — a bar that filters out 95% of feedback.

### Frontend

`frontend/src/components/FeedbackDialog.vue`. Triggered from:

- "Send feedback" link in `HelpView.vue` and `SettingsView.vue` Help section
- Optional small floating "Feedback" button in dev/staging environments only (env-flagged)

Form fields:
- **Type:** Feedback / Bug / Feature request / Question (radio chips)
- **Message:** required textarea
- **Include diagnostics:** checkbox (default on) — appends URL / route / browser / app version / user ID — see Diagnostics section below
- **Email reply-to:** prefilled from logged-in user's email, editable, optional ("don't reply, just listen")

On submit: POST to backend, show success toast, close.

### Backend

```
POST /api/v1/feedback
```

Body: `{type, message, diagnostics, reply_to_email}`. Authenticated users only (anonymous feedback intentionally not in v1 — too easy to abuse without rate-limit + captcha that don't exist yet).

Server side:
- Rate-limit: 5 per hour per user (use the existing rate-limit middleware)
- Forward via the existing email service to `VITE_SUPPORT_EMAIL` / `SUPPORT_EMAIL_TO` env var
- Subject: `[Inventario {Type}] from {user.email}`
- Body: structured plaintext — message + diagnostics + reply-to
- Optionally: also persist to a `feedback_submissions` table for later analytics. **Defer to a follow-up** unless we have a clear consumer.

### Diagnostics payload

When the checkbox is on, the FE includes:
- App version
- Current URL / route name
- Group slug (no IDs)
- User ID, email
- Browser (UA)
- Window size
- Last 10 console errors (if any in-memory log exists; v1 = skip if not)

No automatic screenshot capture — too privacy-sensitive without explicit additional consent.

## Options to decide

1. **Anonymous feedback:** allow `reply_to_email` to be empty and bypass auth? **Default = no, authenticated only.** Anti-abuse.
2. **Issue auto-creation in GitHub:** forward the email + auto-open a GitHub issue with the bug/feature label? **Default = no in v1**, just email. The volume is low enough.
3. **Persist to DB for analytics:** out of v1 scope.
4. **In-app status updates** ("we received your feedback, here's the ticket #"): not in v1.

## Acceptance criteria

- [ ] Dialog reachable from Help/Settings
- [ ] Type/message/diagnostics/reply-to fields
- [ ] Backend endpoint with rate-limit
- [ ] Email forwarded to configured support address
- [ ] e2e: submit feedback, see success toast; backend test asserts email sent

### PR DESCRIPTION
**Pull request by denisvmedia**
Closes #1387.

## Summary

Authenticated users can now submit feedback (general / bug / feature / question) from **Settings → Help → Send feedback**, which lands in the operator-configured support inbox. Per-user rate-limited (5/hour), optional reply-to, optional diagnostics payload.

The route stays mounted in deployments where `SUPPORT_EMAIL` is empty — it returns **503** so the FE can surface the static `mailto:` fallback instead of silently dropping the submission.

## Backend

- **`POST /api/v1/feedback`** — auth-required handler (`go/apiserver/feedback.go`), validates type / message / reply-to, caps the message at 5 KB and diagnostics at 32 lines × 1 KB.
- **`FeedbackRateLimitMiddleware`** + new `AuthRateLimiter.CheckFeedbackAttempt(ctx, userID)` (Redis, in-memory, no-op) — sliding window, 5/hour/user. Per-user keying chosen over per-IP because the surface is auth-gated and CGNAT would otherwise let one bad actor lock unrelated users.
- **`EmailService.SendFeedbackEmail`** added to the interface and to both the `StubEmailService` and `AsyncEmailService`. New `feedback` template (HTML + text) with a dynamic subject built from the submitter's address (`[Inventario {Type}] from {email}`).
- **`SupportEmail`** wired through `bootstrap.Config` → flag (`--support-email`) → `apiserver.Params`. Empty string keeps the route mounted but the handler responds 503.

## Frontend

- **`<FeedbackDialog />`** — radio chips, message textarea, reply-to input (prefilled from `useAuth().user.email`, editable, optional), diagnostics toggle with a live preview of what would be sent.
- **`features/feedback`** slice: `api.ts` + `hooks.ts` (TanStack Query mutation). Tenant-scoped so `skipGroupRewrite: true`.
- 429 surfaces a friendly *"you've sent a lot of feedback recently"* toast; 503 surfaces *"feedback isn't configured on this deployment, please email support directly"*; everything else falls through `parseServerError`.
- Settings → Help row's `ComingSoonBanner` replaced with a `modal:feedback` trigger using the same chrome the shortcuts row already uses.
- `sendFeedback` removed from `coming-soon/registry.ts` and `stubs.json` now that the feature has shipped.
- New `feedback` i18n namespace; `preservePatterns` extended for the dynamic `feedback:types.*` and `feedback:validation.*` keys.

## Tests

- **Go** — `go/apiserver/feedback_test.go` covers happy path, 401, 503-no-support-email, validation (unknown type / empty message / oversize / bad reply-to / malformed JSON), and rate limit (5 accepted → 6th 429 with `Retry-After` + `X-RateLimit-*` headers).
- **vitest** — `frontend/src/components/feedback/__tests__/FeedbackDialog.test.tsx` covers happy submit-and-close, diagnostics-omitted-when-unchecked, validation error on empty message, and the 429 path keeping the dialog open.
- **Playwright** — `e2e/tests/feedback.spec.ts` walks Settings → Help → Send feedback → fill → submit → success toast. `docker-compose.e2e.yaml` + `e2e/setup/setup-stack.ts` set `INVENTARIO_RUN_SUPPORT_EMAIL=support@e2e.test` so the harness exercises the happy path rather than the 503 branch.

## Test plan

- [x] `make swagger` clean (regenerates `go/docs/*` + `frontend/src/types/api.d.ts`)
- [x] `go build ./...`
- [x] `go test ./...` — green
- [x] `golangci-lint run --fix --timeout=5m ./apiserver/... ./services/... ./cmd/...` — 0 issues
- [x] `npm run typecheck` — clean
- [x] `npm run lint` — clean
- [x] `npm test` — 130/130 files, 963 passed + 4 skipped (no regressions; 4 new tests added)
- [x] `npx i18next-cli extract --ci` — clean
- [x] `npm run build` — clean
- [ ] Full Playwright e2e — runs in CI

## Summary by CodeRabbit

* **New Features**
  * In-app Feedback dialog in Settings > Help to send bug reports, feature requests, questions, or general feedback; optional reply-to and diagnostics; success and error toasts.

* **API**
  * New authenticated POST /feedback endpoint to accept submissions.

* **Rate Limiting**
  * Per-user submission limits with appropriate headers and 429 handling.

* **Internationalization**
  * Added English feedback strings; Czech and Russian feedback namespaces initialized.

* **Tests**
  * E2E and unit tests (including screenshots) covering the feedback flow.

* **Chores / Bug Fixes**
  * E2E environment sets stub support email to avoid 503s; stabilized submit-button timing in tests.

[Review Change Stack](https://app.coderabbit.ai/change-stack/denisvmedia/inventario/pull/1734?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: e2e/tests/feedback.spec.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    // and the dialog should close on the 202 response.
+    await dialog.locator('[data-testid="feedback-submit"]').click();
+    await expect(dialog).toBeHidden();
+
+    // Success toast (sonner) confirms the BE accepted the submission.
+    // The toast copy is in en/feedback.json → toasts.success.
+    await expect(page.getByText('Thanks — your feedback was sent.')).toBeVisible();
+    await recorder.takeScreenshot('feedback-02-success-toast');
+  });
+});
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply e2e/tests/feedback.spec.ts:46]: Not adding multi-tenant isolation coverage in this PR — there is nothing tenant-scoped to isolate here.

`POST /api/v1/feedback` is a fire-and-forget pass-through to the operator's single support inbox. The handler doesn't persist anything (the issue brief explicitly defers DB persistence to a follow-up), and the email body itself only carries the submitter's own identity. There is no per-tenant resource a second tenant could observe or fail to observe.

The auth boundary is already exercised by `TestFeedback_Unauthenticated` in `go/apiserver/feedback_test.go` and by the existing `navigateWithAuth` helper used in `feedback.spec.ts`. When/if persistence lands per the "Defer to a follow-up unless we have a clear consumer" note in the issue, a multi-tenant e2e would belong on that PR.

File: frontend/src/components/feedback/FeedbackDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,320 @@
+// FeedbackDialog (issue #1387). In-app channel for the user to file a
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/components/feedback/FeedbackDialog.tsx:1]: Already fixed in b2462f79 — Prettier reformatted FeedbackDialog.tsx (import grouping + line wraps). The latest CI run on `290fad3b` should be green.

File: frontend/src/components/feedback/FeedbackDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+// "valid" text on the FE.
+const MESSAGE_MAX_BYTES = 5 * 1024
+
+const feedbackSchema = z.object({
+  type: z.enum(FEEDBACK_TYPES as unknown as [FeedbackType, ...FeedbackType[]]),
+  message: z
+    .string()
+    .trim()
+    .min(1, "feedback:validation.messageRequired")
+    .max(MESSAGE_MAX_BYTES, "feedback:validation.messageTooLong"),
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/components/feedback/FeedbackDialog.tsx]: Good catch — fixed in 290fad3b.

Replaced `.max(MESSAGE_MAX_BYTES)` with a `.refine` that calls `TextEncoder().encode(value).length`, so the FE bound now matches `len(message)` on the Go side exactly. A multibyte payload (emoji, CJK, accented characters) that previously slipped past Zod's UTF-16 `.max()` would have tripped the BE's 400 — now it fails on the client with the same `messageTooLong` key.

File: frontend/src/components/feedback/FeedbackDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+    }
+  }
+  if (typeof navigator !== "undefined" && navigator.userAgent) {
+    out.user_agent = navigator.userAgent
+  }
+  if (args.userID) out.user_id = args.userID
+  if (args.userEmail) out.user_email = args.userEmail
+  if (args.groupSlug) out.group_slug = args.groupSlug
+  return out
+}
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/components/feedback/FeedbackDialog.tsx:105]: Intentionally out of scope for v1.

The issue brief explicitly says:

> Last 10 console errors (if any in-memory log exists; **v1 = skip if not**)

There is no in-memory console-error buffer in the React frontend today. Wiring one up (a `console.error` wrapper + a ring buffer) is its own design decision (where to mount it, retention bounds, what to do about React's own re-thrown errors in dev mode, etc.) and would inflate this PR well past the AC. Will pick up in a follow-up if the support inbox owner says they actually need it.

File: frontend/src/types/api.d.ts [hunk 5]
```diff
@@ -6260,6 +6340,37 @@ export type components = {
...
+             *     means "the submitter declined to share a reply-to address".
+             */
+            reply_to_email?: string;
+            /**
+             * @description Type is one of "feedback" | "bug" | "feature" | "question". The
+             *     FE renders these as radio chips; the backend uses the value
+             *     verbatim in the email subject and body. Unknown values are
+             *     rejected with 400.
+             */
+            type?: string;
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/types/api.d.ts]: Fixed at the source in 290fad3b — added `validate:"required"` on `Type` / `Message` in `go/apiserver/feedback.go` and `enums:"feedback,bug,feature,question"` on `Type`. The regenerated swagger + `frontend/src/types/api.d.ts` now have `message: string` and `type: "feedback" | "bug" | "feature" | "question"` (no `?`).

File: go/apiserver/feedback.go [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+		// we surface a 503 so the toast can be specific.
+		slog.Warn("Feedback submission rejected: SUPPORT_EMAIL not configured", "user_id", user.ID)
+		http.Error(w, "Feedback is not configured on this deployment", http.StatusServiceUnavailable)
+		return
+	}
+
+	// Limit body to a few KB. The handler reads the body itself
+	// (FeedbackRateLimitMiddleware does not peek), so this is the real
+	// upper bound for the request payload.
+	r.Body = http.MaxBytesReader(w, r.Body, feedbackMaxMessageBytes+8*1024)
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/feedback.go]: Fixed in 290fad3b. Introduced a constant:

```go
const feedbackMaxRequestBodyBytes = feedbackMaxMessageBytes +
    (feedbackMaxDiagnosticsEntries * feedbackMaxDiagnosticsValueBytes) +
    16*1024
```

…and the handler uses it as the `MaxBytesReader` bound. A worst-case-but-valid submission (5 KB message + 32 × 1 KB diagnostics + JSON overhead) now fits comfortably.

File: go/apiserver/feedback.go [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+
+	replyTo := strings.TrimSpace(req.ReplyToEmail)
+	// Loose validation only: the value goes into a plaintext email
+	// body, not into a header on the outbound message, so a malformed
+	// address is at worst noise the inbox owner reads and ignores.
+	// Hard-fail on obvious shape errors (no "@") so a typo is caught
+	// while the user is still in the form.
+	if replyTo != "" && !strings.Contains(replyTo, "@") {
+		http.Error(w, "reply_to_email must be a valid email address", http.StatusBadRequest)
+		return
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/feedback.go]: Fixed in 290fad3b — and the timing was lucky, because in the same commit I also wired the per-message `Reply-To` through to `mailsender.Message.ReplyTo` (per the sibling comment on `email_async_service.go`), so the header-injection hardening is now load-bearing rather than defensive.

The handler now rejects any `\r` / `\n` in `reply_to_email` and parses the address with `net/mail.ParseAddress` instead of a "contains @" sniff:

```go
if replyTo != "" {
    if strings.ContainsAny(replyTo, "\r\n") {
        http.Error(w, "reply_to_email must not contain newlines", http.StatusBadRequest)
        return
    }
    if _, err := mail.ParseAddress(replyTo); err != nil {
        http.Error(w, "reply_to_email must be a valid email address", http.StatusBadRequest)
        return
    }
}
```

File: go/docs/docs.go [hunk 8]
```diff
@@ -5672,6 +5730,38 @@ const docTemplate = `{
...
+                },
+                "reply_to_email": {
+                    "description": "ReplyToEmail is optional. When set the value goes into the email\nbody and (in the async sender) into the Reply-To header. Empty\nmeans \"the submitter declined to share a reply-to address\".",
+                    "type": "string"
+                },
+                "type": {
+                    "description": "Type is one of \"feedback\" | \"bug\" | \"feature\" | \"question\". The\nFE renders these as radio chips; the backend uses the value\nverbatim in the email subject and body. Unknown values are\nrejected with 400.",
+                    "type": "string"
+                }
+            }
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/docs/docs.go:6338]: Fixed at the source. `go/docs/docs.go` is auto-generated by `swag init` — I added `validate:"required"` tags on the Go struct in 290fad3b, regenerated, and the resulting JSON now contains `"required": ["message", "type"]` and an `enum` array on `type`. See the diff on `go/docs/swagger.json`.

File: go/docs/swagger.json [hunk 9]
```diff
@@ -5665,6 +5723,38 @@
...
+                    "type": "string"
+                },
+                "reply_to_email": {
+                    "description": "ReplyToEmail is optional. When set the value goes into the email\nbody and (in the async sender) into the Reply-To header. Empty\nmeans \"the submitter declined to share a reply-to address\".",
+                    "type": "string"
+                },
+                "type": {
+                    "description": "Type is one of \"feedback\" | \"bug\" | \"feature\" | \"question\". The\nFE renders these as radio chips; the backend uses the value\nverbatim in the email subject and body. Unknown values are\nrejected with 400.",
+                    "type": "string"
+                }
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/docs/swagger.json:6330]: Fixed at the source in 290fad3b — added `validate:"required"` on the Go fields and `enums:"feedback,bug,feature,question"` on `Type`, then regenerated. swagger.json now carries both the `required` array and the `enum`, matching the runtime validation in `apiserver/feedback.go`.

File: go/services/email_async_service.go [hunk 10]
```diff
@@ -184,6 +184,22 @@ func (s *AsyncEmailService) SendGroupInviteEmail(ctx context.Context, to, invite
...
+		TemplateType:     emailTemplateFeedback,
+		To:               to,
+		FeedbackType:     feedbackType,
+		FromName:         fromName,
+		FromEmail:        fromEmail,
+		FromUserID:       fromUserID,
+		ReplyToEmail:     replyToEmail,
+		FeedbackMessage:  message,
+		DiagnosticsLines: diagnosticsLines,
+	})
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/services/email_async_service.go:200]: Fixed in 290fad3b. The send path now overrides `s.replyTo` when the job is a feedback submission with a non-empty `ReplyToEmail`:

```go
replyTo := s.replyTo
if job.TemplateType == emailTemplateFeedback {
    if jrt := strings.TrimSpace(job.ReplyToEmail); jrt != "" {
        replyTo = jrt
    }
}
err = s.sender.Send(sendCtx, mailsender.Message{
    To:      job.To,
    From:    s.from,
    ReplyTo: replyTo,
    ...
})
```

Scoped to the feedback template so every other transactional email still falls back to the operator-wide reply-to. The handler's reply-to validation (sibling comment) was tightened in the same commit, so this header path is now safe to feed.

File: go/services/email_service.go [hunk 11]
```diff
@@ -362,6 +377,24 @@ func (s *StubEmailService) SendStorageQuotaWarningEmail(_ context.Context, to, n
...
+	slog.Info("STUB email: feedback submission",
+		"to", to,
+		"from_email", fromEmail,
+		"from_name", fromName,
+		"from_user_id", fromUserID,
+		"feedback_type", feedbackType,
+		"reply_to_email", replyToEmail,
+		"message_chars", len(message),
+		"diagnostics_line_count", len(diagnosticsLines),
+	)
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/services/email_service.go:424]: Fixed in 290fad3b. `StubEmailService.SendFeedbackEmail` no longer logs `from_email`, `from_name`, `from_user_id` or `reply_to_email`. The info log now only carries the operator inbox (`to`), the type, a `has_reply_to` boolean, and the message/diagnostics sizes:

```go
slog.Info("STUB email: feedback submission",
    "to", to,
    "feedback_type", feedbackType,
    "has_reply_to", strings.TrimSpace(replyToEmail) != "",
    "message_chars", len(message),
    "diagnostics_line_count", len(diagnosticsLines),
)
```

The audit trail for "who submitted what" lives in the handler's own `slog.Info("Feedback submitted", "user_id", user.ID, ...)` — that one stays at INFO because the structured `user_id` is already in `auth.LoginEvent`, but PII like the email + name + reply-to are now stripped from the stub sender path.

**Comment by denisvmedia** [comment]
Re: CodeRabbit nitpick — "Add diagnostics boundary cases to the validation table"

`6f35c115` adds the `too_many_diagnostics_lines` case (33 entries → 400) so the documented 32-entry cap now has a guarding test.

The companion `diagnostic_line_too_long` (1025-byte value) case is intentionally **not** a 400. The handler enforces the per-line cap by truncating to `feedbackMaxDiagnosticsValueBytes` and appending `…` in `formatFeedbackDiagnostics` (`go/apiserver/feedback.go:270-271`) — graceful degradation so a long URL or stack frame doesn't reject the whole submission. The entry-count cap is structural (a runaway loop would dump unbounded rows), so that one stays a 400.

If a sender ever needs to *know* a value was truncated, we can revisit by adding a response header like `X-Feedback-Truncated: <count>`, but I don't want to silently expand the 400 surface from the test side without a documented behaviour change.
